Pie percentages switch

alxnddr opened the pull request:
FE piece of https://github.com/metabase/metabase/issues/26776

## Changes

Merge two toggles responsible for showing percentages into a radio group

#### Before 
<img width="364" alt="Screen Shot 2022-11-24 at 1 40 31 AM" src="https://user-images.githubusercontent.com/14301985/203657980-79a49386-e61f-427c-854f-8d540c194fa3.png">

#### After
<img width="367" alt="Screen Shot 2022-11-24 at 1 38 18 AM" src="https://user-images.githubusercontent.com/14301985/203657722-ff743680-e98c-4d11-9c83-e02b1b2c259c.png">

## Screenshots

<img width="1328" alt="Screen Shot 2022-11-24 at 1 37 40 AM" src="https://user-images.githubusercontent.com/14301985/203657658-15348927-ff74-4d80-adb9-a002fc92e618.png">

alxnddr committed:
pie percent switch

npfitz commented:
I noticed when generating this as a static visualization, if you "show percentage in legend" set to true before, it will always show up in the static viz when generated. Here is one where I set the show percentages to "off":

App Viz:
![image](https://user-images.githubusercontent.com/1328979/204386722-4de0a713-e0ec-4fda-a58d-d91365ba8385.png)
Static Viz
![image](https://user-images.githubusercontent.com/1328979/204386599-0e45ad41-e890-496f-9a8a-27fcfa20b255.png)

WiNloSt commented on src/metabase/pulse/render/body.clj:
```diff
@@ -555,7 +555,7 @@
                                         (<= min-a max-b max-a)))]
     (if
-      overlapping-and-valid?
+     overlapping-and-valid?
```
My Calva extension also formats the `if` macro like this, but I checked other places that use this style—if then immediately follows by a line break—the next line should be indented with 2 spaces i.e. reverting this change would make the style more consistent.

WiNloSt reviewed (approved):
FE part looks good to me 👍

alxnddr committed:
review

tsmacdonald committed:
Migrate to card viz-settings v.2 (#26798)

* Migrate to card viz-settings v.2

This uses the new :pie.percent_visibility key (an enum) instead of a
collection of booleans

[#26776]

* Move to def-json-migration

* Don't migrate empty values

alxnddr committed:
Merge branch 'master' into pie-percent

alxnddr committed:
fix specs
